app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -44,9 +60,36 @@ class PopupView : FrameLayout {
     var popup: Popup? = null
         set(value) {
             field = value
-            popupAttributeListAdapter.submitList(getPopupAttributeList())
+            popupAttributeListAdapter.submitList(popupManager.displayedFields)
         }
 
+    var editMode: Boolean = false
+        set(value) {
+            field = value
+            if (field) {
+                popupAttributeListAdapter.submitList(popupManager.editableFields)
+                popupManager.startEditing()
+            } else if (popup != null) {
+                popupAttributeListAdapter.submitList(popupManager.displayedFields)
+                if (!saveEdit) {
```

Comment:
Shouldn't switching the edit mode to false always cancel editing? In other words, if edits should be saved, this should always happen before the edit mode is switched back to false.

Reply:
I have refactored to have the cancel editing logic separate from finish editing and saving popup. This logic is no longer there.